plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java

```diff
@@ -410,11 +358,79 @@ private Iterator<HivePartitionMetadata> getPartitionMetadata(
                 .iterator();
     }
 
-    private TableToPartitionMapping getTableToPartitionMapping(ConnectorSession session, Optional<HiveStorageFormat> storageFormat, SchemaTableName tableName, String partName, List<Column> tableColumns, List<Column> partitionColumns)
+    private static HivePartitionMetadata toPartitionMetadata(
```

Comment:
Can we do the refactoring of extracting this method in a separate commit ?

Reply:
Unfortunately this commit essentially is just this method extraction already, the changes are all related but semantically just move the bulk of the logic that was previously inline inside the lambda into this static method (which requires some additional parameters, etc).